mock/py/mockbuild/package_manager.py

```diff
@@ -293,7 +296,24 @@ def initialize(self):
         self.copy_certs()
         if self.buildroot.is_bootstrap:
             self.copy_distribution_gpg_keys()
-        self.initialize_config()
+        config_content = self.initialize_config()
+
+        config = ConfigParser()
+        config.read_file(StringIO(config_content))
+
+        for section in config.sections():
+            if 'baseurl' not in config[section]:
+                continue
+            srcdir = config[section]['baseurl'].split('file://')[-1]
+            util.mkdirIfAbsent(srcdir)
```

Comment:
You probably don't want to create the srcdir.  If the directory doesn't exist, it's an user error.